src/server.h

```diff
@@ -404,6 +406,12 @@ extern int configOOMScoreAdjValuesDefaults[CONFIG_OOM_COUNT];
 #define CLIENT_REPROCESSING_COMMAND (1ULL<<50) /* The client is re-processing the command. */
 #define CLIENT_PREREPL_DONE (1ULL<<51) /* Indicate that pre-replication has been done on the client */
 
+#define CLIENT_REPL_MAIN_CHANNEL (1ULL<<52) /* RDB Channel: track a connection
+                                                 which is used for online replication data */
+#define CLIENT_REPL_RDB_CHANNEL (1ULL<<53) /* RDB Channel: track a connection
+                                                 which is used for rdb snapshot */
+#define CLIENT_PROTECTED_RDB_CHANNEL (1ULL<<54) /* Client should kept until PSYNC establishment. */
```

Comment:
I have gone through the references to CLIENT_PROTECTED_RDB_CHANNEL a few times but I still don't think I understand its purpose. More specifically, what could be the problem if this client is freed before PSYNC establishment is completed? What could lead to such a client failing to complete PSYNC handshake?